kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/inband/DecryptionDekCache.java

```diff
@@ -90,7 +92,8 @@ public DecryptionDekCache(@NonNull DekManager<K, E> dekManager,
      * @param executor The executor
      * @return A future
      */
-    private CompletableFuture<Dek<E>> loadDek(CacheKey<E> cacheKey, Executor executor) {
+    @RunsOnThread("#DecryptionDekCache(dekCacheExecutor)")
```

Comment:
hmm thats an interesting case for keeping it as a string.

Reply:
It's tempting to specify some expression language for being able to describe what we mean more precisely. While I think that would be really interesting (in the academic sense), especially in tandem with some kind of enforcement, it's not our goal. 

That said, I've got no problem with tweaking how we're describing the threads to make this documentation more useful/helpful.